spec/models/reservation_spec.rb

```diff
@@ -87,705 +73,830 @@
-  context 'with equipment item/model matching problems' do
-    subject(:reservation) do
-      r = FactoryGirl.build(:valid_reservation)
-      r.equipment_item = FactoryGirl.create(:equipment_item)
-      r
+  describe '#duration' do
+    it 'returns the length of the reservation' do
+      res = FactoryGirl.build_stubbed(:valid_reservation,
+                                      start_date: Time.zone.today,
+                                      due_date: Time.zone.today + 1.day)
+      length = 2
+      expect(res.duration).to eq(length)
     end
+  end
 
-    it { is_expected.not_to be_valid }
-    it 'should not save' do
-      expect(reservation.save).to be_falsey
-      expect(Reservation.all.size).to eq(0)
+  describe '#time_checked_out' do
+    it 'returns the length of the checkout for returned reservations' do
+      res = FactoryGirl.build_stubbed(:checked_in_reservation,
+                                      start_date: Time.zone.today - 3.days,
+                                      due_date: Time.zone.today - 1.day,
+                                      checked_out: Time.zone.today - 3.days,
+                                      checked_in: Time.zone.today - 2.days)
+      length = 2
+      expect(res.time_checked_out).to eq(length)
     end
-    it 'cannot be updated' do
-      reservation.start_date = Time.zone.today + 1.day
-      expect(reservation.save).to be_falsey
+  end
+
+  describe '#late_fee' do
+    it 'returns the correct late fee' do
+      fee_per_day = 5
+      days = 3
+      expected = fee_per_day * days
+      model = FactoryGirl.build_stubbed(:equipment_model, late_fee: fee_per_day)
+      res = FactoryGirl.build_stubbed(:overdue_reservation,
+                                      equipment_model: model,
+                                      due_date: Time.zone.today - days.days)
+      expect(res.late_fee).to eq(expected)
     end
-    it 'fails appropriate validations' do
-      expect(reservation.matched_item_and_model).not_to be_nil
+    it 'returns 0 if not overdue' do
+      res = FactoryGirl.build_stubbed(:checked_out_reservation)
+      expect(res.late_fee).to eq(0)
     end
-    it 'passes other custom validations' do
-      expect(reservation.start_date_before_due_date).to be_nil
-      expect(reservation.not_empty).to be_nil
-      expect(reservation.not_in_past).to be_nil
-      expect(reservation.check_banned).to be_nil
-      expect(reservation.validate).to eq([])
+    it 'returns the cap if a cap is set' do
+      fee_per_day = 5
+      days = 3
+      cap = 10
+      model = FactoryGirl.build_stubbed(:equipment_model,
+                                        late_fee: fee_per_day,
+                                        late_fee_max: cap)
+      res = FactoryGirl.build_stubbed(:overdue_reservation,
+                                      equipment_model: model,
+                                      due_date: Time.zone.today - days.days)
+      expect(res.late_fee).to eq(model.late_fee_max)
```

Comment:
This is a minor point, but would it be better if our assertion was against `cap` instead of `model.late_fee_max`? I understand that they're equal but `cap` is the value we assigned at the beginning (i.e. our input) so it seems like we should be checking against that.